python/ray/air/_internal/remote_storage.py

```diff
@@ -298,7 +298,7 @@ def upload_to_uri(
         )
 
     if not exclude:
-        _ensure_directory(bucket_path)
+        _ensure_directory(uri)
```

Comment:
This can also just call `_ensure_directory(bucket_path, fs)` to avoid reparsing?